[DOCS] GX Cloud Architecture Supporting Content (#11439)
great-expectations/great_expectations: docs/docusaurus/docs/reference/learn/gx_in_your_data_pipeline/transformation.md

---
sidebar_label: 'Transformation'
title: 'Transformation'

> ---

Reviewer: This page should have a `description:` in the frontmatter for SEO

Data analysts often face the challenge of ensuring that aggregate data remains consistent as it flows through different transformation layers. It’s also critical to standardize records effectively during these transformations to maintain accuracy and comparability across datasets. Additionally, identifying and addressing errant or anomalous data points—whether by removing them or flagging them for further review—is an ongoing concern.

> ![Sample data set showing the progression of data through the medallion architecture.](/img/integration-medallion.png)

Reviewer: This alt text is repetitive of the surrounding prose. I suggest making it more descriptive to deduplicate

```suggestion
![Example data warehouse with bronze layer to cleanse tables, silver layer to combine tables, and gold layer to calculate aggregates.](/img/integration-medallion.png)
```

> *This sample data set shows the progression of data through the medallion architecture.*

Reviewer: As far as I know we don't use this pattern of italics after a screenshot to signal a caption anywhere else. I suggest removing this styling from all descriptions of screenshots.

> *This sample data set shows the progression of data through the medallion architecture.*

Reviewer: This use of the term "data set" feels off to me. I think of each table in the diagram as a data set. 

```suggestion
This sample data pipeline shows the progression of data through the medallion architecture.
```

### How can GX Cloud help solve these problems?

> GX Cloud can be implemented effectively within a medallion architecture to help surface data quality issues at each transformation stage. For example, consider bronze layer subscription data consisting of 1,000 rows. If 100 of these rows contain shipping dates that are earlier than their transaction dates, this indicates a problem. GX Cloud allows you to create Expectations that flag such inconsistencies automatically, alerting various teams within the organization. The analytics team can be alerted to investigate potential transformation issues. Meanwhile, although data ingestion may have been technically complete and accurate from a raw data standpoint, the sales team may need to correct the records at the source, since some issues may not be discovered until transformation has occurred. Implementing GX Cloud within the medallion framework enables proactive detection and resolution of such issues before they impact downstream analysis or reporting.

Reviewer: The word "effectively" seems unnecessary here and feels a bit repetitive since it's already used in the first paragraph of this page. I suggest removing it for concision

```suggestion
GX Cloud can be implemented within a medallion architecture to help surface data quality issues at each transformation stage. For example, consider bronze layer subscription data consisting of 1,000 rows. If 100 of these rows contain shipping dates that are earlier than their transaction dates, this indicates a problem. GX Cloud allows you to create Expectations that flag such inconsistencies automatically, alerting various teams within the organization. The analytics team can be alerted to investigate potential transformation issues. Meanwhile, although data ingestion may have been technically complete and accurate from a raw data standpoint, the sales team may need to correct the records at the source, since some issues may not be discovered until transformation has occurred. Implementing GX Cloud within the medallion framework enables proactive detection and resolution of such issues before they impact downstream analysis or reporting.
```

> GX Cloud can be implemented effectively within a medallion architecture to help surface data quality issues at each transformation stage. For example, consider bronze layer subscription data consisting of 1,000 rows. If 100 of these rows contain shipping dates that are earlier than their transaction dates, this indicates a problem. GX Cloud allows you to create Expectations that flag such inconsistencies automatically, alerting various teams within the organization. The analytics team can be alerted to investigate potential transformation issues. Meanwhile, although data ingestion may have been technically complete and accurate from a raw data standpoint, the sales team may need to correct the records at the source, since some issues may not be discovered until transformation has occurred. Implementing GX Cloud within the medallion framework enables proactive detection and resolution of such issues before they impact downstream analysis or reporting.

Reviewer: The example

> For example, consider bronze layer subscription data consisting of 1,000 rows. If 100 of these rows contain shipping dates that are earlier than their transaction dates, this indicates a problem. GX Cloud allows you to create Expectations that flag such inconsistencies automatically, alerting various teams within the organization.

feels a bit odd to me because it doesn't match what in the diagram that follows. I suggest editing this to be more related to the diagram

> ![Examples of data moving from the bronze to silver layers.](/img/integration-transformation.png)

Reviewer: I think this alt text is insufficiently descriptive. The image contains information about Expectations that is not conveyed in the alt text

suggestion to make this more descriptive

```suggestion
![Expectations on the bronze layer such as "Expect column values to be between" and "Expect column values to be of type" help clean data before it gets combined in the silver layer. Custom SQL Expectations on the silver layer test that data adheres to business requirements.](/img/integration-transformation.png)
```

### How can GX Cloud help solve these problems?

> GX Cloud can be implemented within a medallion architecture to help surface data quality issues at each transformation stage. For example, consider bronze layer subscription data. GX Cloud allows you to create Expectations that flag errors and inconsistencies, such as typos and differences in data types. Expectations can also be set up to detect logical errors between two columns, like if the shipping date occurs before the payment date. The analytics team can be alerted to investigate potential transformation issues. Meanwhile, although data ingestion may have been technically complete and accurate from a raw data standpoint, the sales team may need to correct the records at the source, since some issues may not be discovered until transformation has occurred. Implementing GX Cloud within the medallion framework enables proactive detection and resolution of such issues before they impact downstream analysis or reporting.

Reviewer: Thanks for updating the example to be more related to the diagram that follows. 

That being said, the flow of this paragraph feels a bit off to me
- the first sentence says "help surface data quality issues at each transformation stage" but then only the bronze layer is explicitly mentioned
- "Expectations can also be set up to detect logical errors between two columns, like if the shipping date occurs before the payment date. " describes something illustrated at the silver layer, but the user's mental context is still "consider bronze layer subscription data" at this point. 

Here's one suggestion for improving the flow:

```suggestion
GX Cloud can be implemented within a medallion architecture to help surface data quality issues at each transformation stage. For example, consider a bronze layer for cleansing data. GX Cloud allows you to create Expectations that flag errors and inconsistencies, such as typos and differences in data types. Next, consider a silver layer for combining data. Expectations can be set up to detect logical errors between columns in a combined silver table that couldn't be detected across disparate bronze tables, like if the shipping date occurs before the payment date. In both cases, the analytics team can be alerted to investigate potential transformation issues. Meanwhile, although data ingestion may have been technically complete and accurate from a raw data standpoint, the sales team may need to correct the records at the source, since some issues may not be discovered until transformation has occurred. Implementing GX Cloud within the medallion framework enables proactive detection and resolution of such issues before they impact downstream analysis or reporting.
```
